src/modules/authorization/utils.ts

```diff
@@ -16,6 +16,37 @@ export function hasAuthorization(
   )
 }
 
+export function hasAuthorizationAndEnoughAllowance(
+  authorizations: Authorization[],
+  authorizationToFind: Authorization,
+  allowance: string
```

Comment:
What if we change the name of this variable to:
```suggestion
  requiredAllowance: string
```